File: 1475.final-prices-with-a-special-discount-in-a-shop.js
/*
 * @lc app=leetcode id=1475 lang=javascript
 *
 * [1475] Final Prices With a Special Discount in a Shop
 */

// @lc code=start
/**
 * @param {number[]} prices
 * @return {number[]}
 */
var finalPrices = function(prices) {
  return prices.map((v,i)=>{
    let discount = prices.slice(i+1).find(v2=>v2<=v) || 0
    return v - discount
  })
};
// @lc code=end

File: 1475.final-prices-with-a-special-discount-in-a-shop.py
#
# @lc app=leetcode id=1475 lang=python3
#
# [1475] Final Prices With a Special Discount in a Shop
#

# @lc code=start
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        for i in range(0,len(prices)):
            for j in range(i+1,len(prices)):
                if prices[j] <= prices[i]:
                    prices[i] -= prices[j]
                    break

        return prices
# @lc code=end

